### PR TITLE
Ubuntu Linux 20.04 rework

### PR DESCRIPTION
**Pull request by 72nomada**
|Related issue|
|---|
|#14690|

This PR will rebuild the current Ubuntu Linux 20.04 policy using the  CIS Ubuntu Linux 20.04 LTS Benchmark v1.1.0 - 03-31-2021

### CODE REVIEW

**Comment by 72nomada** [comment]
Items ready for team review: 

| Id | Title | Team Review status |
|---|---|---|
| 1 | Initial Setup ||
| 1.1 | Filesystem Configuration  ||
| 1.1.1 | Disable unused filesystems ||
| 1.1.1.1 | Ensure mounting of cramfs filesystems is disabled (Automated)  ||
| 1.1.1.2 | Ensure mounting of freevxfs filesystems is disabled (Automated)  ||
| 1.1.1.3 | Ensure mounting of jffs2 filesystems is disabled (Automated) ||
| 1.1.1.4 | Ensure mounting of hfs filesystems is disabled (Automated)  ||
| 1.1.1.5 | Ensure mounting of hfsplus filesystems is disabled (Automated) ||
| 1.1.1.6 | Ensure mounting of squashfs filesystems is disabled (Manual)  ||
| 1.1.1.7 | Ensure mounting of udf filesystems is disabled (Automated) ||
| 1.1.2 | Ensure /tmp is configured (Automated)  ||
| 1.1.3 | Ensure nodev option set on /tmp partition (Automated)  ||
| 1.1.4 | Ensure nosuid option set on /tmp partition (Automated)  ||
| 1.1.5 | Ensure noexec option set on /tmp partition (Automated)  ||
| 1.1.6 | Ensure /dev/shm is configured (Automated)  ||
| 1.1.7 | Ensure nodev option set on /dev/shm partition (Automated)  ||
| 1.1.8 | Ensure nosuid option set on /dev/shm partition (Automated)  ||
| 1.1.9 | Ensure noexec option set on /dev/shm partition (Automated) ||
| 1.1.10 | Ensure separate partition exists for /var (Automated)  ||
| 1.1.11 | Ensure separate partition exists for /var/tmp (Automated)||
| 1.1.12 | Ensure /var/tmp partition includes the nodev option (Automated)  ||
| 1.1.13 | Ensure /var/tmp partition includes the nosuid option (Automated)  ||
| 1.1.14 | Ensure /var/tmp partition includes the noexec option (Automated)||
| 1.1.15 | Ensure separate partition exists for /var/log (Automated)||

File: ruleset/sca/ubuntu/cis_ubuntu20-04.yml [hunk 1]
```diff
@@ -345,9 +375,10 @@ checks:
       - http://tldp.org/HOWTO/LVM-HOWTO/
     condition: all
     rules:
-      - 'c:mount -> r:\s/var/log\s'
+      - 'c:findmnt -n /var/log -> r:\s*/var/log\s'
 
-  - id: 19020
+  # 1.1.16 Ensure separate partition exists for /var/log/audit (Automated)
+  - id: 19021
```
Comment (fabamatic) [inline ruleset/sca/ubuntu/cis_ubuntu20-04.yml:381]: Repeated id with 1.1.17

**Review by fabamatic — DISMISSED** [review]

File: ruleset/sca/ubuntu/cis_ubuntu20-04.yml [hunk 2]
```diff
@@ -23,95 +23,115 @@ requirements:
...
     compliance:
       - cis: ["1.1.1.1"]
       - cis_csc: ["5.1"]
       - pci_dss: ["2.2.5"]
       - tsc: ["CC6.3"]
     condition: all
     rules:
-      - 'c:modprobe -n -v cramfs -> r:^install /bin/true'
-      - 'not c:lsmod -> r:cramfs'
+      - "c:modprobe -n -v cramfs -> r:^install /bin/true|^install /bin/false"
```
[the remark this answers is not in the document]
Comment (72nomada) [reply ruleset/sca/ubuntu/cis_ubuntu20-04.yml:50]: Both true and false are valid options

File: ruleset/sca/ubuntu/cis_ubuntu20-04.yml [hunk 3]
```diff
@@ -23,95 +23,115 @@ requirements:
...
     description: "The cramfs filesystem type is a compressed read-only Linux filesystem embedded in small footprint systems. A cramfs image can be used without having to first decompress the image."
     rationale: "Removing support for unneeded filesystem types reduces the local attack surface of the server. If this filesystem type is not needed, disable it."
-    remediation: "1) Edit or create a file in the /etc/modprobe.d/ directory ending in .conf and add the following line: install cramfs /bin/true. 2) Run the following command to unload the cramfs module: # rmmod cramfs"
+    remediation: "Edit or create a file in the /etc/modprobe.d/ directory ending in .conf and add the following line: install cramfs /bin/true. Run the following command to unload the cramfs module: # rmmod cramfs."
     compliance:
       - cis: ["1.1.1.1"]
       - cis_csc: ["5.1"]
       - pci_dss: ["2.2.5"]
       - tsc: ["CC6.3"]
     condition: all
```
Comment (fabamatic) [inline ruleset/sca/ubuntu/cis_ubuntu20-04.yml:48]: Condition may be any instead of all?

File: ruleset/sca/ubuntu/cis_ubuntu20-04.yml [hunk 4]
```diff
@@ -23,95 +23,115 @@ requirements:
...
     description: "The freevxfs filesystem type is a free version of the Veritas type filesystem. This is the primary filesystem type for HP-UX operating systems."
     rationale: "Removing support for unneeded filesystem types reduces the local attack surface of the system. If this filesystem type is not needed, disable it."
-    remediation: "Edit or create a file in the /etc/modprobe.d/ directory ending in .conf Example: vi /etc/modprobe.d/freevxfs.conf and add the following line: install freevxfs /bin/true Run the following command to unload the freevxfs module: # rmmod freevxfs"
+    remediation: "Edit or create a file in the /etc/modprobe.d/ directory ending in .conf Example: vi /etc/modprobe.d/freevxfs.conf and add the following line: install freevxfs /bin/true. Run the following command to unload the freevxfs module: # rmmod freevxfs."
     compliance:
       - cis: ["1.1.1.2"]
       - cis_csc: ["5.1"]
       - pci_dss: ["2.2.5"]
       - tsc: ["CC6.3"]
     condition: all
```
Comment (fabamatic) [inline ruleset/sca/ubuntu/cis_ubuntu20-04.yml:64]: Condition may be any instead of all?

**Comment by 72nomada** [comment]
|Check ID | Check Name | implemented | Ready for review |
|---|---|---|---|---|
|1.2 | Configure Software Updates  | ⚫  | |  
|1.2.1 | Ensure package manager repositories are configured (Manual)  | 🔴  |Not implemented |  
|1.2.2 | Ensure GPG keys are configured (Manual)  | 🔴  | Not implemented |  
|1.3 | Filesystem Integrity Checking | ⚫   | |  
|1.3.1 | Ensure AIDE is installed (Automated)  | 🟢  | |  
|1.3.2 | Ensure filesystem integrity is regularly checked (Automated)  |  | |  
|1.4 | Secure Boot Settings  |  | |  
|1.4.1 | Ensure permissions on bootloader config are not overridden (Automated) |  | |  
|1.4.2 | Ensure bootloader password is set (Automated) |  | |  
|1.4.3 | Ensure permissions on bootloader config are configured (Automated) |  | |  
|1.4.4 | Ensure authentication required for single user mode (Automated)  |  | |  
|1.5 | Additional Process Hardening  |  | |  
|1.5.1 | Ensure XD/NX support is enabled (Manual) |  | |  
|1.5.2 | Ensure address space layout randomization (ASLR) is enabled (Automated)  |  | |  
|1.5.3 | Ensure prelink is not installed (Automated) |  | |  
|1.5.4 | Ensure core dumps are restricted (Automated) |  | |

**Comment by jmv74211** [comment]
## QA report

The corresponding checks between 1 and 1.9 have been tested and approved (after making the modifications requested in the review).

- https://github.com/wazuh/wazuh-qa/issues/3431
- https://github.com/wazuh/wazuh-qa/issues/3432
- https://github.com/wazuh/wazuh-qa/issues/3433

The rest of the checks will be reviewed as they are implemented, with an estimated date of October 28th.

File: ruleset/sca/ubuntu/cis_ubuntu20-04.yml [hunk 5]
```diff
@@ -23,2024 +23,2407 @@ requirements:
...
+      - "c:dpkg -s iptables -> r:package 'iptables' is not installed"
+      - "c:dpkg -s ufw -> r:package 'ufw' is not installed"
+
+  # 3.5.2.2 Ensure ufw is uninstalled or disabled with nftables (Automated)
+  - id: 19099
+    title: "Ensure ufw is uninstalled or disabled with nftables."
+    description: "Uncomplicated Firewall (UFW) is a program for managing a netfilter firewall designed to be easy to use."
+    rationale: "Running both the nftables service and ufw may lead to conflict and unexpected results."
+    remediation: "Run one of the following commands to either remove ufw or disable ufw Run the following command to remove ufw: # apt purge ufw Run the following command to disable ufw: # ufw disable"
     com
```
Comment (fabamatic) [inline ruleset/sca/ubuntu/cis_ubuntu20-04.yml:2324]: Shouldn't it be:
`f:/boot/grub/grub.cfg -> r:^\s*linux && r:audit=1'` ?

**Comment by jmv74211** [comment]
## QA report

The corresponding checks between 3.5 and 4.4 have been tested and approved (after making the modifications requested in the review).

- https://github.com/wazuh/wazuh-qa/issues/3436
- https://github.com/wazuh/wazuh-qa/issues/3438

**Comment by jmv74211** [comment]
## QA report

The corresponding checks between 5 to 5.3.22 have been tested and approved (after making the modifications requested in the review).

- https://github.com/wazuh/wazuh-qa/issues/3439

**Review by fabamatic — CHANGES_REQUESTED** [review]

**Comment by jmv74211** [comment]
## QA report

The [Ubuntu Linux 20.04 SCA Policy rework end to end test](https://github.com/wazuh/wazuh-qa/issues/3520) has passed successfully.

**Review by jmv74211 — APPROVED** [review]
## QA review

- **Type**: Manual testing.
- **Status**: Approved  🟢 
- **Testing issue**: https://github.com/wazuh/wazuh-qa/issues/3391
- **Comments**:  https://github.com/wazuh/wazuh-qa/issues/3391#issuecomment-1308615655

**Pull request merged**